test/BaseTestEntityWithSources.cpp

```diff
@@ -82,4 +82,30 @@ void BaseTestEntityWithSources::testSourceVectorSetter() {
     std::vector<Source> deleter;
     da.sources(deleter);
     CPPUNIT_ASSERT(da.sourceCount() == 0);
-}
\ No newline at end of file
+}
+
+void BaseTestEntityWithSources::testSourceOrder() {
+    std::vector<std::string> names = { "source_a", "source_b", "source_c", "source_d", "source_e" };
```

Comment:
One thing: maybe we should randomise that order (maybe even in code, with `std::random_shuffle`). I am actually not entirely convinced that this wouldn't break the existing code. But it is late and I might be wrong.